File: C/ShellSort.c
#include <stdio.h>

void swap(int *x,int *y)

{

    int temp=*x;

    *x=*y;

    *y=temp;

}

void shellsort(int arr[],int n)

{

    int gap,i,j;

    for(gap=n/2;gap>=1;gap=gap/2)

    {

        for(j=gap;j<n;j++)

        {

            for(i=j-gap;i>=0;i=i-gap)

            {

                if(arr[i+gap]>arr[i])

                {

                    break;

                }

                else

                {

                    swap(&arr[i+gap],&arr[i]);

                }

            }

        }

    }

}

int main (void)

{

   int i;

    int arr[]={3,7,9,10,12,6,5,2,1,18};

    int n=10;

    shellsort(arr,n);

    printf("After Sorting...\n");

    

    for(i=0;i<n;i++)

    {

        printf(" %d ",arr[i]);

    }

}

File: JavaScript/shellSort.js
let customGap = [700, 300, 132, 57, 23, 10, 4 , 1]
function shellSortWithCustomGaps(arr) {
	let n = arr.length;

	//Start with a really large gap, and then reduce the gap until there isn't any
	//This is to use a custom gap array that you setup beforehand.
	for (let gapIndex = 0; gapIndex < customGap.length; gapIndex++)
	{
		let gap = customGap[gapIndex];
		//Do a insertion sort for each of the section the gap ends up dividing
		for (let i = gap; i < n; i += 1)
		{
			//We store the current varible
			let temp = arr[i];
			
			//This is the insection sort to sort the section into order
			let j;
			for (j = i; j >= gap && arr[j-gap] > temp; j-=gap)
			{
				arr[j] = arr[j-gap];
			}

			arr[j] = temp;
		}
	}

	return arr;
}